### PR TITLE
Add support for TIMEOUT key for cmd tests ##r2r

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Right now the timeout was a global defined in Config. This PR adds support for per-test timeout, in case some tests are expected to take a bit more time.

**Test plan**

`db/formats/pe/65535sects` should pass in ASAN build in travis.

### CODE REVIEW

**Comment by ret2libc** [comment]
Ok, it seems it worked well https://travis-ci.com/github/radareorg/radare2/builds/182614378#L3208 .

File: binr/r2r/r2r.h [hunk 1]
```diff
@@ -37,6 +37,12 @@ typedef struct r2r_cmd_test_bool_record {
 	bool set;
 } R2RCmdTestBoolRecord;
 
+typedef struct r2r_cmd_test_num_record {
+	ut64 value;
+	ut64 line; // bools are always oneliners (e.g. TIMEOUT=10)
```
Comment (kazarmy) [inline binr/r2r/r2r.h:42]:
```suggestion
	ut64 line; // nums are always oneliners (e.g. TIMEOUT=10)
```

**Review by kazarmy — CHANGES_REQUESTED** [review]

File: binr/r2r/load.c [hunk 2]
```diff
@@ -198,17 +198,43 @@ R_API RPVector *r2r_load_cmd_test_file(const char *file) {
...
+				*cmt = '\0'; \
+				cmt--; \
+				while (cmt > val && *cmt == ' ') { \
+					*cmt = '\0'; \
+					cmt--; \
+				} \
+			} \
+			char *endval; \
+			test->field.value = strtol (val, &endval, 0); \
+			if (!endval || *endval) { \
```
Comment (kazarmy) [inline binr/r2r/load.c:227]: This condition is a bit strange ... https://linux.die.net/man/3/strtol says:

"If endptr is not NULL, strtol() stores the address of the first invalid character in *endptr. If there were no digits at all, strtol() stores the original value of nptr in *endptr (and returns 0). In particular, if *nptr is not '\0' but **endptr is '\0' on return, the entire string is valid."

I think `endval` here is pointing to the newline at the end.

Comment (kazarmy) [reply binr/r2r/load.c:227]: Oh sorry misread the manual -- the first condition was just to check whether `endval` is valid. My bad.

**Review by kazarmy — APPROVED** [review]

**Review by thestr4ng3r — APPROVED** [review]
wtf are you dpoing, you could just change one line and it was done

**Pull request merged**